### PR TITLE
check for empty CSV file

### LINKED ISSUE #3: Add in check for empty prediction file generated
@Simon-Harris-IBM The log for 9700807 you sent me had errors:

```
File size must be at least one bytes
```

The workflow failed on the upload step because you cannot upload files into Synapse if it is empty

Submission 9700804 generated a prediction file that was invalid.  The `check_status` tool intentionally quits the workflow.  In the log you can see:
```
STDERR: 2020-03-02T04:56:53.149231447Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    17   if(inputs.status == "INVALID"){
STDERR: 2020-03-02T04:56:53.149313839Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    18     throw 'invalid submission';
STDERR: 2020-03-02T04:56:53.149406601Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    19   } else {
STDERR: 2020-03-02T04:56:53.149489540Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    20     return {finished: true};
STDERR: 2020-03-02T04:56:53.149577906Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    21   }
STDERR: 2020-03-02T04:56:53.149659129Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    22 })()
STDERR: 2020-03-02T04:56:53.149750582Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    stdout was: ''
STDERR: 2020-03-02T04:56:53.149832871Z 12504aee8408 2020-03-02 04:56:53,149 MainThread WARNING toil.leader: kind-https_raw.githubusercontent.com_Sage-Bionetworks_ChallengeWorkflowTemplates_v2.1_check_status.cwl/instancem4rggvk6    stderr was: 'invalid submission'
```

### PR DESCRIPTION
**Pull request by mummert-ibm**
Require CSV file to contain at least one row

### CODE REVIEW

**Comment by thomasyu888** [comment]
We will have to add something in run_docker.py for this to work.

File: run_docker.py [hunk 1]
```diff
@@ -198,6 +198,9 @@ def main(syn, args):
         # expects there to be a folder named /output to write to
         raise Exception("No 'predictions.csv' file written to /output, "
                         "please check inference docker")
+    elif os.stat(os.path.join(output_dir, 'predictions.csv')).st_size == 0:
+        raise Exception("'predictions.csv' file is empty.  Please check inference docker")
```
Comment (thomasyu888) [inline run_docker.py]: Don't want to raise the exception. We actually want to create a prediction file when one can't be found.  This way we can hide the toil cwl log files from participants

**Review by thomasyu888 — CHANGES_REQUESTED** [review]

File: run_docker.py (hunk 1 above)
Comment (mummert-ibm) [reply run_docker.py]: Not sure why empty or missing files isn't handled in the file loader.    The validate_and_score.py code has always checked for a missing file, and is now checking for files that have no rows (empty).      Therefore, it's results.json will give an appropriate reason for the submission being invalid.
'''
mummert@ubntu-v100-x2-todd:scoring$ ./validate_and_score.py -f empty -a $HOME/objectnet/scoring/answers_by_id.json -r
{
  "prediction_file_errors": [
    "Output CSV file contained no valid rows"
  ],
  "prediction_file_status": "INVALID"
}
mummert@ubntu-v100-x2-todd:scoring$ ./validate_and_score.py -f does_not_exist -a $HOME/objectnet/scoring/answers_by_id.json -r
{
  "prediction_file_errors": [
    "Unable to open results file: does_not_exist"
  ],
  "prediction_file_status": "INVALID"
}
'''
Do you want run_docker.py to never generate an exception?

**Review by thomasyu888 — COMMENTED** [review]
The main reason is cwl expects a prediction file output, that's why it must exist.  If the docker container fails, the prediction file isn't created , therefore it would cause cwl to fail.  The alternative is to have an optional output for run_docker.cwl tool

**Comment by thomasyu888** [comment]
@Simon-Harris-IBM I made edits to this. I stopped throwing an exception in `run_docker.py` and created an empty prediction file if it doesn't exist.

That being said, synapse doesn't allow the upload of empty files, so I added a custom upload tool with a try catch that will catch when a synapse upload doesn't occur. (Note, I do realize that I'm catching a broad exception here, but all errors from the client are usually `SynapseHTTPErrors` so it doesn't make much of a difference.

Then this will go onto the validation step where participants will be notified of their errors.

**Pull request merged**